### ISSUE TITLE
How to create custom endpoint and add it to the documentation?

### ISSUE THREAD

**Issue by neluzhin** [api-platform/api-platform#352]
I want to create endpoint like `/api/v1/me` that returns information (`User` object) about currently authenticated user and add it to my documentation. 

I have an action:

````php
namespace AppBundle\Action\Me;

use AppBundle\Entity\User;
use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;

class MeView
{

    /**
     * @var TokenStorageInterface
     */
    private $tokenStorage;

    public function __construct(TokenStorageInterface $tokenStorage)
    {
        $this->tokenStorage = $tokenStorage;
    }

    /**
     * @Security("is_authenticated()")
     *
     * @Route(
     *     name="me_view",
     *     path="/me",
     *     methods={"GET"},
     *     defaults={"_api_item_operation_name"="view_myself"}
     * )
     *
     * @return User
     */
    public function __invoke()
    {
        return $this->tokenStorage->getToken()->getUser();
    }
}
````

I define this action in my entity:

````php
namespace AppBundle\Entity;

use ApiPlatform\Core\Annotation as Api;
use AppBundle\Repository\UserRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @Api\ApiResource(
 *     itemOperations={
 *         "view_myself"={"route_name"="me_view"}
 *     }
 * )
 *
 * @ORM\Entity(repositoryClass=UserRepository::class)
 */
class User
{

    /**
     * @ORM\Id()
     * @ORM\GeneratedValue(strategy="AUTO")
     * @ORM\Column(type="integer")
     *
     * @var int
     */
    private $id;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(?int $id)
    {
        $this->id = $id;
    }
}

````
But in the docs it asks for `id` parameter.

![image](https://user-images.githubusercontent.com/12474739/27840120-37739c06-60ff-11e7-9ea9-57aa4c93fba8.png)

Didn't found in Api Platform docs same recipes. What am I doing wrong?

**Comment by Gounlaf** [comment]
Maybe this: #311 ?

**Comment by ssimplix** [comment]
@Gounlaf, this one about endpoints mapped with entities.

**Comment by aalinia** [comment]
I think your action should have something like this:
`
defaults={"_api_resource_class"=User::class, "_api_collection_operation_name"="view_myself"}
`

**Comment by Gounlaf** [comment]
@ssimplix I figured how to fix it (thx to #311)

```php
namespace AppBundle\Entity;

use ApiPlatform\Core\Annotation as Api;
use AppBundle\Repository\UserRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @Api\ApiResource(
 *     itemOperations={
 *         "view_myself"={
 *             "route_name"="me_view",
 *             "swagger_context"={
 *                  "parameters"={}
 *              }
 *         }
 *     }
 * )
 *
 * @ORM\Entity(repositoryClass=UserRepository::class)
 */
```
